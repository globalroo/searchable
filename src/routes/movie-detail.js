import React from "react";
import { Divider, Typography } from "@material-ui/core";

import { AssetScroller } from "src/components/asset-scroller";
import { CastMembers } from "src/components/cast-members";
import { getMovieById } from "src/tmdb-service/tmdb-api";
import { getSmallPosterImage } from "src/tmdb-service/tmdb-config";
import { useByIdLoader } from "../hooks/by-id-loader";

import "./movie-detail-style.css";

export const MovieDetail = ({ id }) => {
	const { loading, error, response } = useByIdLoader({ id, fetcher: getMovieById });
	const { payload } = response;

	//TODO: Make Loading and Error components...
	if (loading) return <div>Loading</div>;
	if (error) return <div>error</div>;

	//TODO: Break out into smaller components
	return (
		<div>
			{payload ? (
				<div className="wrapper">
					<aside className="sidebar">
						<img src={getSmallPosterImage(payload.poster_path)} alt={payload.title} />
					</aside>
					<article className="content">
						<h1 className="bold-header-uppercase">{payload.title}</h1>
						<h1>{payload.tagline}</h1>
						<div className="content-row-1">
							<Typography>{payload.overview}</Typography>
						</div>
						<Divider />
						<div className="content-row-2">
							<CastMembers cast={payload.credits.cast} />
						</div>
						<Divider />
					</article>
					{payload.recommendations.total_results > 0 && (
						<footer className="footer">
							<h1>Recommendations</h1>
							<AssetScroller assets={payload.recommendations.results} />
						</footer>
					)}
				</div>
			) : null}
		</div>
	);
};
