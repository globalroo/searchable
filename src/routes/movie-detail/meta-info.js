import React from "react";
import { Divider, Typography } from "@material-ui/core";
import { CastMembers } from "src/components/cast-members";

export const MetaInfo = ({ tagline, overview, title, cast }) => (
	<article className="content">
		<h1 className="bold-header-uppercase">{title}</h1>
		<h1>{tagline}</h1>
		<div className="content-row-1">
			<Typography>{overview}</Typography>
		</div>
		<Divider />
		<div className="content-row-2">
			<CastMembers cast={cast} />
		</div>
		<Divider />
	</article>
);
