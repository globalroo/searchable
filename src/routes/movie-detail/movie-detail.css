.content {
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    overflow: hidden;
}

.content-row-1 {
    flex: 1;
    align-items: flex-start;
    justify-content: flex-start;
    flex-grow: 1
}

.content-row-2 {
    align-items: flex-end;
    justify-content: flex-end;
    overflow: hidden;
}