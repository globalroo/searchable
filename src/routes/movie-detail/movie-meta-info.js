import React from "react";
import { PersonScroller } from "src/components/person-scroller";
import { NavigateBackContainer } from "src/components/navigate-back";
import { Divider, Typography } from "@material-ui/core";

import { format } from "date-fns";

export const MetaInfo = ({ tagline, overview, title, cast, release_date }) => (
	<article className="content">
		<h1 className="bold-header-uppercase">
			<NavigateBackContainer>
				{title} [{format(release_date, "YYYY")}]
			</NavigateBackContainer>
		</h1>
		<h1>{tagline}</h1>
		<div className="content-row-1">
			<Typography>{overview}</Typography>
		</div>
		<Divider />
		<div className="content-row-2">
			<PersonScroller cast={cast} />
		</div>
		<Divider />
	</article>
);
