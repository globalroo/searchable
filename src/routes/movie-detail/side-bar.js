import React from "react";
import { useImageLoader } from "src/hooks/image-loader";
import { getSmallPosterImage } from "src/tmdb-service/tmdb-config";

export const SideBar = ({ poster_path, title }) => {
	const [image] = useImageLoader(getSmallPosterImage(poster_path));
	return (
		<aside className="sidebar">
			<img src={image} alt={title} />
		</aside>
	);
};
