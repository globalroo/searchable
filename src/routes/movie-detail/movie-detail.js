import React from "react";

import "./movie-detail.css";
import { AssetScroller } from "src/components/asset-scroller";
import { getMovieById } from "src/tmdb-service/tmdb-api";
import { MEDIA_TYPE } from "../../tmdb-service/tmdb-api";
import { MetaInfo } from "./movie-meta-info";
import { SideBar } from "src/components/side-bar";
import { useByIdLoader } from "src/hooks/by-id-loader";

export const MovieDetail = ({ id }) => {
	const { loading, error, response } = useByIdLoader({
		id,
		fetcher: getMovieById
	});

	const { payload } = response;

	if (loading) return null;
	if (error) return <div>Error</div>;
	if (!payload) return null;

	return (
		<div className="wrapper">
			<SideBar poster_path={payload.poster_path} title={payload.title} />
			<MetaInfo
				title={payload.title}
				release_date={payload.release_date}
				tagline={payload.tagline}
				overview={payload.overview}
				cast={payload.credits.cast}
			/>
			{payload.recommendations.total_results > 0 && (
				<footer className="footer">
					<h1>Recommendations</h1>
					<AssetScroller
						assets={payload.recommendations.results}
						link_type={MEDIA_TYPE.MOVIE}
					/>
				</footer>
			)}
		</div>
	);
};
