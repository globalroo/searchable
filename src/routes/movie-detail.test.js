import React from "react";
import { render } from "react-testing-library";

import { MovieDetail } from "./movie-detail";

const mockMovie = {
	adult: false,
	backdrop_path: "/vmrDTWGTupJQ8KRUNrx4RLGtTVp.jpg",
	belongs_to_collection: null,
	budget: 20000000,
	genres: [{ id: 35, name: "Comedy" }],
	homepage: null,
	id: 226486,
	imdb_id: "tt2103254",
	original_language: "en",
	original_title: "Tammy",
	overview:
		"After losing her job and learning that her husband has been unfaithful, a woman hits the road with her profane, hard-drinking grandmother.",
	popularity: 8.93,
	poster_path: "/cq6wvDqETqJXgpQplkL0FBw2leM.jpg",
	production_companies: [
		{ id: 12, logo_path: "/iaYpEp3LQmb8AfAtmTvpqd4149c.png", name: "New Line Cinema", origin_country: "US" }
	],
	production_countries: [{ iso_3166_1: "US", name: "United States of America" }],
	release_date: "2014-07-02",
	revenue: 100525432,
	runtime: 97,
	spoken_languages: [{ iso_639_1: "en", name: "English" }],
	status: "Released",
	tagline: "She hit the road. The road hit back.",
	title: "Tammy",
	video: false,
	vote_average: 5.2,
	vote_count: 700,
	recommendations: {
		page: 1,
		results: [
			{
				adult: false,
				backdrop_path: "/2wKw8M6g7wj0RNz9zAeqU9s1BKl.jpg",
				genre_ids: [28, 35, 80],
				id: 136795,
				original_language: "en",
				original_title: "The Heat",
				overview:
					'Uptight and straight-laced, FBI Special Agent Sarah Ashburn is a methodical investigator with a reputation for excellence--and hyper-arrogance. Shannon Mullins, one of Boston P.D.\'s "finest," is foul-mouthed and has a very short fuse, and uses her gut instinct and street smarts to catch the most elusive criminals. Neither has ever had a partner, or a friend for that matter. When these two wildly incompatible law officers join forces to bring down a ruthless drug lord, they become the last thing anyone expected: Buddies.',
				poster_path: "/tM3hDt8JcU8Y8i6vzTcbnocM2bx.jpg",
				release_date: "2013-06-27",
				title: "The Heat",
				video: false,
				vote_average: 6.6,
				vote_count: 2340,
				popularity: 12.497
			}
		],
		total_pages: 1,
		total_results: 1
	},
	credits: {
		cast: [
			{
				cast_id: 4,
				character: "Tammy",
				credit_id: "52fe4e7f9251416c75158661",
				gender: 1,
				id: 55536,
				name: "Melissa McCarthy",
				order: 0,
				profile_path: "/nRNMJlqR33j84cGdB0RxstV3NYm.jpg"
			}
		],
		crew: [
			{
				credit_id: "52fe4e7f9251416c75158651",
				department: "Directing",
				gender: 2,
				id: 170820,
				job: "Director",
				name: "Ben Falcone",
				profile_path: "/abhsMUOZKyAGPK3uZPdf3UQ5MuZ.jpg"
			}
		]
	}
};

jest.mock("src/hooks/by-id-loader", () => ({
	useByIdLoader: () => ({ response: { payload: mockMovie }, loading: false, error: false })
}));

describe("Movie Detail tests", () => {
	it("Renders a movie detail component", () => {
		const { container } = render(<MovieDetail id={mockMovie.id} />);
		expect(container.firstChild).toMatchSnapshot();
	});
});
