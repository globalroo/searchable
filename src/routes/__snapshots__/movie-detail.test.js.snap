// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie Detail tests Renders a movie detail component 1`] = `
<div>
  <div
    class="wrapper"
  >
    <aside
      class="sidebar"
    >
      <img
        alt="Tammy"
        src="http://image.tmdb.org/t/p/w300//cq6wvDqETqJXgpQplkL0FBw2leM.jpg"
      />
    </aside>
    <article
      class="content"
    >
      <h1
        class="bold-header-uppercase"
      >
        Tammy
      </h1>
      <h1>
        She hit the road. The road hit back.
      </h1>
      <div
        class="content-row-1"
      >
        <p
          class="MuiTypography-root-1 MuiTypography-body1-10"
        >
          After losing her job and learning that her husband has been unfaithful, a woman hits the road with her profane, hard-drinking grandmother.
        </p>
      </div>
      <hr
        class="MuiDivider-root-37"
      />
      <div
        class="content-row-2"
      >
        <ul
          class="MuiGridList-root-45 ImageAvatars-gridList-44"
          style="margin: -2px;"
        >
          <div
            class="[object Object]"
            style="width: 50%; height: 184px; padding: 2px;"
          >
            <a
              href="/person/55536"
            >
              <div
                class="MuiAvatar-root-46 ImageAvatars-bigAvatar-42"
              >
                <img
                  alt="Melissa McCarthy"
                  class="MuiAvatar-img-48"
                  src="http://image.tmdb.org/t/p/w300//nRNMJlqR33j84cGdB0RxstV3NYm.jpg"
                />
              </div>
            </a>
            <p
              class="MuiTypography-root-1 MuiTypography-body1-10 MuiTypography-alignCenter-24"
            >
              Melissa McCarthy
            </p>
            <span
              class="MuiTypography-root-1 MuiTypography-caption-11 MuiTypography-alignCenter-24"
            >
              Tammy
            </span>
          </div>
        </ul>
      </div>
      <hr
        class="MuiDivider-root-37"
      />
    </article>
    <footer
      class="footer"
    >
      <h1>
        Recommendations
      </h1>
      <div
        class="SingleLineGridList-root-49"
      >
        <ul
          class="MuiGridList-root-45 SingleLineGridList-gridList-50"
          style="margin: -2px;"
        >
          <a
            href="/movie/136795"
            style="width: 40%; height: 184px; padding: 2px;"
          >
            <li
              class="MuiGridListTile-root-53"
            >
              <div
                class="MuiGridListTile-tile-54"
              >
                <img
                  alt="The Heat"
                  src="http://image.tmdb.org/t/p/w300//2wKw8M6g7wj0RNz9zAeqU9s1BKl.jpg"
                />
                <div
                  class="MuiGridListTileBar-root-57 MuiGridListTileBar-titlePositionBottom-58"
                >
                  <div
                    class="MuiGridListTileBar-titleWrap-61 MuiGridListTileBar-titleWrapActionPosRight-63"
                  >
                    <div
                      class="MuiGridListTileBar-title-64"
                    >
                      The Heat
                    </div>
                  </div>
                  <div
                    class="MuiGridListTileBar-actionIcon-66"
                  >
                    <button
                      class="MuiButtonBase-root-74 MuiIconButton-root-68"
                      tabindex="0"
                      type="button"
                    >
                      <span
                        class="MuiIconButton-label-73"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root-77 SingleLineGridList-title-51"
                          focusable="false"
                          role="presentation"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
                          />
                          <path
                            d="M0 0h24v24H0z"
                            fill="none"
                          />
                        </svg>
                      </span>
                      <span
                        class="MuiTouchRipple-root-86"
                      />
                    </button>
                  </div>
                </div>
              </div>
            </li>
          </a>
        </ul>
      </div>
    </footer>
  </div>
</div>
`;
